### PR TITLE
[ACL-228] Add Net9/Net8 support

### PR DESCRIPTION
**Pull request by tl-Roberto-Mancinelli**
- Added Net9/Net8 support
- Upgrade CI to latest version of cake/github actions
- Added AuthorizationScope const
- Removed moq dependency in test

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -14,31 +14,25 @@ on:
...
+      - uses: actions/checkout@v4
+      - name: Setup .NET 9.0
+        uses: actions/setup-dotnet@v2
         with:
-          fetch-depth: 0
-      - name: Setup Java
-        uses: actions/setup-java@v1
+          dotnet-version: 9.0.x
+      - name: Setup .NET 8.0
+        uses: actions/setup-dotnet@v2
```
[the remark this answers is not in the document]
Comment (tl-Roberto-Mancinelli) [reply .github/workflows/build.yml:25]: I think net9 would be enough, I just followed the previous convention

File: examples/MvcExample/Controllers/HomeController.cs [hunk 2]
```diff
@@ -142,10 +142,10 @@ void SetProviderAndSchemeId(OneOf<PaymentMethod.BankTransfer, PaymentMethod.Mand
...
             return apiResponse.Data.Match(
                 authRequired => Failed(authRequired.Status, authRequired.PaymentMethod),
-                authorizing => SuccessOrPending(authorizing),
-                authorized => SuccessOrPending(authorized),
-                success => SuccessOrPending(success),
-                settled => SuccessOrPending(settled),
+                SuccessOrPending,
+                SuccessOrPending,
+                SuccessOrPending,
+                SuccessOrPending,
```
[the remark this answers is not in the document]
Comment (tl-Roberto-Mancinelli) [reply examples/MvcExample/Controllers/HomeController.cs:148]: I like this syntax as it's more compact but as you said it's just an example, I'm okay with both

File: src/TrueLayer/Mandates/Model/Beneficiary.cs [hunk 3]
```diff
@@ -1,8 +1,3 @@
-using System;
```
[the remark this answers is not in the document]
Comment (tl-Roberto-Mancinelli) [reply src/TrueLayer/Mandates/Model/Beneficiary.cs:1]: absolutely, adding a tech debt ticket for that

File: examples/MvcExample/Controllers/HomeController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dili91) [reply examples/MvcExample/Controllers/HomeController.cs:148]: No strong preferences from my side, let's keep it as you're suggesting!

**Review by dili91 — DISMISSED** [review]

**Pull request merged**